### ISSUE TITLE
Implement torch.sum for FP32

### ISSUE THREAD

**Issue by ashwindk-ibm** [torch-spyre/torch-spyre#1871]
### What feature would you like to be added?

The sum reduction operation should run in FP32 in torch-spyre.

### Why is this feature needed?

The sum reduction operation should run in FP32 in torch-spyre.

**Comment by Sunny-Anand** [comment]
@moriohara @yoheiueda @michihirohorie I think we can pick this issue if no one else is working to check if we can enable this with FP32 amax enablement which would enable softmax to use upstream softmax. If so pls assign to me.

**Comment by lupalby** [comment]
This should be working from a backend point of view, is there any pending problem?

**Comment by moriohara** [comment]
@lupalby Is torch.sum supposed to work for unaligned fp32 tensors as well for staggered fp32 tensors?

**Comment by lupalby** [comment]
What are unaligned fp32 tensors? Staggered tensors (the result of fp16->fp32 upcasting) are supported.
